lib/mutant/ast/regexp/transformer.rb

```diff
@@ -0,0 +1,82 @@
+module Mutant
+  module AST
+    module Regexp
+      class Transformer
+        include AbstractType
+
+        REGISTRY = {} # rubocop:disable MutableConstant
```

Comment:
@backus although I am not entirely sure I understand what the separation between `Registry` and `TypeRegistry` is in the example above. Given that `Registry` has no other state besides a `TypeRegistry` instance I'm not sure why you don't collapse them down and have a single `Registry` object that you interact with.